### ISSUE TITLE
Improve error messages when missing UMLS data

### ISSUE THREAD

**Issue by ddomingof** [pyobo/pyobo#71]
UMLS still requires manual intervention for loading. However, it's still possible to request names from UMLS CURIEs, which leads to unhelpful error messages like below.

**Solution**: Improve error messages and explanation on how users can get the data themselves

```python-traceback
>>> import pyobo
>>> pyobo.get_name_from_curie('UMLS:C0156404')
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
    _main()
  File "/~/venv/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/~/venv/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/~/venv/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/~/venv/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
    indication_name = pyobo.get_name_by_curie(curated_curie)
  File "/~/venv/lib/python3.8/site-packages/pyobo/extract.py", line 57, in get_name_by_curie
    return get_name(prefix, identifier)
  File "/~/venv/lib/python3.8/site-packages/pyobo/extract.py", line 63, in get_name
    id_name = get_id_name_mapping(prefix)
  File "/~/venv/lib/python3.8/site-packages/pyobo/extract.py", line 88, in get_id_name_mapping
    return _get_id_name_mapping()
  File "/~/venv/lib/python3.8/site-packages/pyobo/cache_utils.py", line 48, in _wrapped
    rv = f()
  File "/~/venv/lib/python3.8/site-packages/pyobo/extract.py", line 84, in _get_id_name_mapping
    obo = get(prefix, **kwargs)
  File "/~/venv/lib/python3.8/site-packages/pyobo/getters.py", line 52, in get
    obo.write_default()
  File "/~/venv/lib/python3.8/site-packages/pyobo/struct/struct.py", line 314, in write_default
    self.write(path, use_tqdm=use_tqdm)
  File "<decorator-gen-756>", line 2, in write
  File "/~/venv/lib/python3.8/site-packages/networkx/utils/decorators.py", line 240, in _open_file
    result = func_to_be_decorated(*new_args, **kwargs)
  File "/~/venv/lib/python3.8/site-packages/pyobo/struct/struct.py", line 297, in write
    for line in it:
  File "/~/venv/lib/python3.8/site-packages/pyobo/struct/struct.py", line 288, in iterate_obo_lines
    for term in self.iter_terms():
  File "/~/venv/lib/python3.8/site-packages/pyobo/sources/umls/umls.py", line 74, in iter_terms
    with zipfile.ZipFile(path) as zip_file:
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/zipfile.py", line 1268, in __init__
    self._RealGetContents()
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/zipfile.py", line 1335, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
```

**Comment by cthoyt** [comment]
Need more context on this one. What code did you run? UMLS is a huge pain, because I basically had to sign my life away to get access to the data in the first place

**Comment by ddomingof** [comment]
I used pyobo.get_name_from_curie() I can send you the curie to reproduce the error

**Comment by ddomingof** [comment]
pyobo.get_name_from_curie('UMLS:C0156404')

**Comment by cthoyt** [comment]
Closed with 852479b
